### PR TITLE
Refactor CloudMachineClient and related types into new package

### PR DESCRIPTION
**Pull request by christothes**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/cloudmachine/Azure.CloudMachine/api/Azure.CloudMachine.netstandard2.0.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+namespace Azure.Template
```
Comment (KrzysztofCwalina) [inline sdk/cloudmachine/Azure.CloudMachine/api/Azure.CloudMachine.netstandard2.0.cs:1]: I don't think we want this, do we?

**Review by KrzysztofCwalina — CHANGES_REQUESTED** [review]

File: sdk/cloudmachine/Azure.CloudMachine/api/Azure.CloudMachine.netstandard2.0.cs (hunk 1 above)
Comment (christothes) [reply sdk/cloudmachine/Azure.CloudMachine/api/Azure.CloudMachine.netstandard2.0.cs]: Forgot to re-export. It should be fixed now

**Review by KrzysztofCwalina — APPROVED** [review]

**Comment by christothes** [comment]
@microsoft-github-policy-service rerun

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -180,6 +180,12 @@
 # ServiceLabel: %Bot Service
 # ServiceOwners:                                                   @sgellock
 
+# PRLabel: %CloudMachine
```
Comment (jsquire) [inline .github/CODEOWNERS:183]:
```suggestion
# PRLabel: %Cloud Machine
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -180,6 +180,12 @@
...
+/sdk/cloudmachine/                                                 @christothes @KrzysztofCwalina
+
+# ServiceLabel: %ProvCloudMachineisioning
```
Comment (jsquire) [inline .github/CODEOWNERS:186]:
```suggestion
# ServiceLabel: %Cloud Machine
```

**Review by jsquire — APPROVED** [review]
CODEOWNERS changes look good with the label update to match `Cloud Machine` - which I just created.   This may take 24 hours to have the linter pick it up, so we can bypass the linter if this is ready before.

**Pull request merged**